test/unit_test/components/movement_sensor_test.dart

```diff
@@ -0,0 +1,285 @@
+import 'package:viam_sdk/src/resource/manager.dart';
+import 'package:viam_sdk/src/utils.dart';
+import 'package:viam_sdk/viam_sdk.dart';
+
+class FakeMovementSensor extends MovementSensor {
+  Map<String, dynamic>? extra;
+
+  @override
+  String name;
+
+  FakeMovementSensor(this.name);
+
+  @override
+  Future<Map<String, dynamic>> doCommand(Map<String, dynamic>? command) async {
+    return {'command': command};
+  }
+
+  @override
+  Future<Map<String, double>> accuracy({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return {'accuracy': 0};
+  }
+
+  @override
+  Future<Vector3> angularVelocity({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<double> compassHeading({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return 0;
+  }
+
+  @override
+  Future<Vector3> linearAcceleration({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<Vector3> linearVelocity({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Vector3(x: 0);
+  }
+
+  @override
+  Future<Orientation> orientation({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Orientation(oX: 0);
+  }
+
+  @override
+  Future<Position> position({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Position(GeoPoint(latitude: 0), 0);
+  }
+
+  @override
+  Future<Properties> properties({Map<String, dynamic>? extra}) async {
+    this.extra = extra;
+    return Properties(angularVelocitySupported: false);
+  }
+}
+
+void main() {
+  group('FakeMovementSensor Tests', () {
+    const String name = 'movementSensor';
+    late FakeMovementSensor movementSensor;
+
+    setUp(() {
+      movementSensor = FakeMovementSensor(name);
+    });
+
+    test('accuracy', () async {
+      final result = await movementSensor.accuracy();
+      expect(result, {'accuracy': 0});
+    });
+
+    test('angularVelocity', () async {
+      final result = await movementSensor.angularVelocity();
+      expect(result.x, 0);
+    });
+
+    test('compassHeading', () async {
+      final result = await movementSensor.compassHeading();
+      expect(result, 0);
+    });
+
+    test('linearAcceleration', () async {
+      final result = await movementSensor.linearAcceleration();
+      expect(result.x, 0);
+    });
+
+    test('linearVelocity', () async {
+      final result = await movementSensor.linearVelocity();
+      expect(result.x, 0);
+    });
+
+    test('orientation', () async {
+      final result = await movementSensor.orientation();
+      expect(result.oX, 0);
+    });
+
+    test('position', () async {
+      final result = await movementSensor.position();
+      expect(result.altitude, 0);
+    });
+
+    test('properties', () async {
+      final result = await movementSensor.properties();
+      expect(result.angularVelocitySupported, false);
```

Comment:
Im assuming this is setting the parameter of angularVelocitySupported to false

Reply:
> why false instead of 0?

in this case it is looking for a bool and not a double, so i am checking that it is false, which is the expected result.

> Im assuming this is setting the parameter of angularVelocitySupported to false

this is not setting it, but it is getting returned as false in the Fake class. This is testing to see that it returns false, which is the expected result.